api/src/main/scala/Geoprocessing.scala

```diff
@@ -0,0 +1,65 @@
+package org.wikiwatershed.mmw.geoprocessing
+
+import java.util.concurrent.atomic.LongAdder
+
+import collection.concurrent.TrieMap
+
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+
+import geotrellis.raster._
+import geotrellis.raster.rasterize._
+import geotrellis.vector._
+import geotrellis.vector.io._
+
+import geotrellis.spark._
+
+trait Geoprocessing extends Utils {
+  /**
+    * For an InputJson object, return a histogram of raster grouped count results.
+    *
+    * @param   input  The InputData
+    * @return         A histogram of results in JSON format
+    */
+  def getRasterGroupedCount(input: InputData) = {
+    val aoi = createAOIFromInput(input)
+    val rasterLayers = createRasterLayerIds(input.rasters, input.zoom, aoi)
+    Map("result" -> rasterGroupedCount(rasterLayers, aoi)).toJson
```

Comment:
Instead of returning  `Map(...).toJson`, we should make a case class for the result type (which is the same for all endpoints) and use that instead:

```diff
diff --git a/api/src/main/scala/Geoprocessing.scala b/api/src/main/scala/Geoprocessing.scala
index 2a0a40c..1fc2777 100644
--- a/api/src/main/scala/Geoprocessing.scala
+++ b/api/src/main/scala/Geoprocessing.scala
@@ -21,10 +21,11 @@ trait Geoprocessing extends Utils {
     * @param   input  The InputData
     * @return         A histogram of results in JSON format
     */
-  def getRasterGroupedCount(input: InputData) = {
+  def getRasterGroupedCount(input: InputData): ResultJson = {
     val aoi = createAOIFromInput(input)
     val rasterLayers = createRasterLayerIds(input.rasters, input.zoom, aoi)
-    Map("result" -> rasterGroupedCount(rasterLayers, aoi)).toJson
+
+    Result(rasterGroupedCount(rasterLayers, aoi))
   }
 
   /**
diff --git a/api/src/main/scala/WebServer.scala b/api/src/main/scala/WebServer.scala
index 432fb5a..b377eb4 100644
--- a/api/src/main/scala/WebServer.scala
+++ b/api/src/main/scala/WebServer.scala
@@ -13,10 +13,12 @@ case class InputData(operationType: String, rasters: List[String], zoom: Int,
   polygonCRS: String, rasterCRS: String, polygon: List[String])
 
 case class PostRequestJson(input: InputData)
+case class Result(result: Map[String, Int])
 
 object PostRequestJsonProtocol extends DefaultJsonProtocol {
   implicit val inputFormat = jsonFormat6(InputData)
   implicit val postFormat = jsonFormat1(PostRequestJson)
+  implicit val resultFormat = jsonFormat1(Result)
 }
 
 object WebServer extends HttpApp with App with LazyLogging with Geoprocessing {
```